tests/virtualization/universal/storage.pm

```diff
@@ -25,64 +27,28 @@ sub run_test {
-    assert_script_run "virsh pool-define-as testing dir - - - - '/pool_testing'";
-
-    record_info "Pool list";
-    assert_script_run "virsh pool-list --all | grep testing";
-
-    record_info "Pool build";
-    assert_script_run "virsh pool-build testing";
-
-    record_info "Pool start";
-    assert_script_run "virsh pool-start testing";
-
-    record_info "Pool autostart";
-    assert_script_run "virsh pool-autostart testing";
-
-    record_info "Pool info";
-    assert_script_run "virsh pool-info testing";
-
-    record_info "Create";
-    assert_script_run("virsh vol-create-as testing $_-storage 100M", 120) foreach (keys %virt_autotest::common::guests);
-
-    record_info "Listing";
-    assert_script_run("ls /pool_testing/", 120) foreach (keys %virt_autotest::common::guests);
-    assert_script_run("virsh vol-list testing | grep $_-storage", 120) foreach (keys %virt_autotest::common::guests);
-
-    record_info "Info";
-    assert_script_run("virsh vol-info --pool testing $_-storage", 120) foreach (keys %virt_autotest::common::guests);
-
-    record_info "Dump XML";
-    assert_script_run("virsh vol-dumpxml --pool testing $_-storage", 120) foreach (keys %virt_autotest::common::guests);
-
-    record_info "Resize";
-    assert_script_run("virsh vol-resize --pool testing $_-storage 200M", 120) foreach (keys %virt_autotest::common::guests);
-
-    record_info "Attached";
-    my $target = (is_xen_host) ? "xvdx" : "vdx";
-    assert_script_run("virsh attach-disk --domain $_ --source `virsh vol-path --pool testing $_-storage` --target $target", 120) foreach (keys %virt_autotest::common::guests);
-    assert_script_run("virsh detach-disk $_ $target", 120) foreach (keys %virt_autotest::common::guests);
+    assert_script_run "virsh pool-define-as $dir_pool_name dir - - - - '/pool_testing'";
 
-    record_info "Clone";
-    assert_script_run("virsh vol-clone --pool testing $_-storage $_-clone", 120) foreach (keys %virt_autotest::common::guests);
-    assert_script_run("virsh vol-info --pool testing $_-clone", 120) foreach (keys %virt_autotest::common::guests);
+    # Dir Storage Pool Basic Management
+    virt_storage_pool_management($dir_pool_name);
 
-    record_info "Remove";
-    assert_script_run("virsh vol-delete --pool testing $_-clone", 120) foreach (keys %virt_autotest::common::guests);
-    assert_script_run("virsh vol-delete --pool testing $_-storage", 120) foreach (keys %virt_autotest::common::guests);
+    # Dir Storage Volume Basic Management
+    my $dir_vol_size = '100M';
+    my $dir_vol_resize = '200M';
+    virt_storage_vol_management($dir_pool_name, size => $dir_vol_size, dir => 1, resize => $dir_vol_resize);
 
-    record_info "Pool destroy";
-    assert_script_run "virsh pool-destroy testing";
-    assert_script_run "virsh pool-delete testing";
-    assert_script_run "virsh pool-undefine testing";
+    ## Cleanup
+    # Destroy the Dir storage pool
+    destroy_virt_storage_pool($dir_pool_name);
 }
```

Comment:
What did you do here ? Just rearrange test code and package them in virtual_storage_utils.pm ?

Reply:
refer to func `destroy_virt_storage_pool ()`
```
sub destroy_virt_storage_pool {
    my $vstorage_pool_name = shift;
    record_info "Storage Pool destroy";
    assert_script_run "virsh pool-destroy $vstorage_pool_name";
    assert_script_run "virsh pool-delete $vstorage_pool_name";
    assert_script_run "virsh pool-undefine $vstorage_pool_name";
}
```

after finish virt storage pool test, need to call func `destroy_virt_storage_pool ()` to clean up test environment. 
currently, this func `destroy_virt_storage_pool ()` support Virt Storage pool with Dir and LVM together.